gateway/gateway.go

```diff
@@ -39,6 +39,11 @@ type API interface {
 	// exist due to a missing link, it should return an error of type:
 	// https://pkg.go.dev/github.com/ipfs/go-path@v0.3.0/resolver#ErrNoLink
 	ResolvePath(context.Context, path.Path) (path.Resolved, error)
+
+	// HasDNSLinkRecord returns if the provided path has a DNSLink TXT record.
+	// It does not perform any validation, only checks for the existence of
+	// a DNSLink TXT record.
+	HasDNSLinkRecord(context.Context, string) bool
```

Comment:
(as noted in https://github.com/ipfs/kubo/pull/9624#discussion_r1096844466)

I feel this will not only have more utility, but will match GetIPNSRecord more closely.

```suggestion
	// GetDNSLinkRecord returns if the provided FQDN has a DNSLink TXT record.
	// It does not perform recursive resolution like ResolvePath, only checks for the existence of
	// DNSLink TXT record with path starting with /ipfs/ or /ipns/ and returns the path as-is.
	GetDNSLinkRecord(context.Context, string) (path.Path, error)
```